core/node/consistency_checker/src/lib.rs

```diff
@@ -380,52 +376,36 @@ impl ConsistencyChecker {
     const DEFAULT_SLEEP_INTERVAL: Duration = Duration::from_secs(5);
 
     pub async fn new(
-        l1_client: Box<DynClient<L1>>,
-        gateway_client: Option<Box<DynClient<L1>>>,
+        sl_client: Box<dyn EthInterface>,
         max_batches_to_recheck: u32,
         pool: ConnectionPool<Core>,
         commitment_mode: L1BatchCommitmentMode,
-        l2_chain_id: L2ChainId,
+        settlement_layer: SettlementLayer,
     ) -> anyhow::Result<Self> {
         let (health_check, health_updater) = ConsistencyCheckerHealthUpdater::new();
-        let l1_chain_id = l1_client.fetch_chain_id().await?;
-        let l1_chain_data = SLChainAccess {
-            client: l1_client.for_component("consistency_checker"),
-            chain_id: l1_chain_id,
+        let sl_chain_id = sl_client.fetch_chain_id().await?;
+        let chain_data = SLChainAccess {
+            client: sl_client,
+            chain_id: sl_chain_id,
             diamond_proxy_addr: None,
         };
 
-        let gateway_chain_data = if let Some(client) = gateway_client {
-            let gateway_diamond_proxy =
-                CallFunctionArgs::new("getZKChain", Token::Uint(l2_chain_id.as_u64().into()))
-                    .for_contract(L2_BRIDGEHUB_ADDRESS, &bridgehub_contract())
-                    .call(&client)
-                    .await?;
-            let chain_id = client.fetch_chain_id().await?;
-            Some(SLChainAccess {
-                client: client.for_component("consistency_checker"),
-                chain_id,
-                diamond_proxy_addr: Some(gateway_diamond_proxy),
-            })
-        } else {
-            None
-        };
         Ok(Self {
             contract: zksync_contracts::hyperchain_contract(),
             max_batches_to_recheck,
             sleep_interval: Self::DEFAULT_SLEEP_INTERVAL,
-            l1_chain_data,
-            gateway_chain_data,
+            chain_data,
+            settlement_layer,
             event_handler: Box::new(health_updater),
-            l1_data_mismatch_behavior: L1DataMismatchBehavior::Log,
+            l1_data_mismatch_behavior: L1DataMismatchBehavior::Bail,
```

Comment:
With this change, is the `Log` variant (= the enum as a whole) still necessary? Previously, `Bail` was a test-only variant used to ensure that consistency failures lead to failing unit tests.

Reply:
yes, it was one of the questions to you (as an author), why it was like this and could we remove it?

Reply:
Ah OK. I've explained why (for unit tests), so it should be safe to remove now.